update extensions documentation for Rancher 2.9 updates (#11362)
rancher/dashboard: docusaurus/docs/extensions/support-matrix.md

# Support matrixes

## Shell support matrix

> The Shell package is the core package that allows for an Extension to incorporate with a Rancher Dashboard.  

Reviewer: ```suggestion
The Shell package enables Extensions to integrate with Rancher.
```
